include/deal.II/dofs/dof_accessor.h

```diff
@@ -310,6 +327,13 @@ class DoFAccessor : public dealii::internal::DoFAccessorImplementation::
   operator=(const DoFAccessor<structdim, dim, spacedim, level_dof_access> &da) =
     delete;
 
+  /**
+   * Move assignment operator.
+   */
+  DoFAccessor<structdim, dim, spacedim, level_dof_access> &
+  operator=(DoFAccessor<structdim, dim, spacedim, level_dof_access>
```

Comment:
Does this work with older versions of clang? I expect that we would need a `NOLINT` here.

Reply:
We currently have a bunch of places in the library with `noexcept` for move constructor/assignment operator. I suspect the `noexcept` problem triggers when a std container calls `move` but it's just a guess. Not sure if it matters if we are using `libc++` or `libstc++`. The CI OSX did trigger the bug in my PR, so I was hoping that it would catch it again.